whd/dispatcher.py

```diff
@@ -139,6 +145,89 @@ def _pipeline_definition_changed(self, push_event):
             return True
         return False
 
+    def determine_affected_pipelines(self, push_event) -> Pipeline:
+        '''yield each concourse pipeline that may be affected by the given push-event.
+        '''
+        repo_enumerator = concourse.enumerator.GithubRepositoryDefinitionEnumerator(
+            repository_url=push_event.repository().repository_url(),
+            cfg_set=self.cfg_set,
+        )
+
+        for descriptor in repo_enumerator.enumerate_definition_descriptors():
+            # need to merge and consider the effective definition
+            effective_definition = descriptor.pipeline_definition
+            for override in descriptor.override_definitions:
+                effective_definition = ci.util.merge_dicts(effective_definition, override)
+
+            yield Pipeline(
+                pipeline_name=descriptor._pipeline_name(),
```

Comment:
I did not check the entire PR, yet. do you also use this new method in the replication-codepath? If not, please do